lib/boxmaker.scad

```diff
@@ -0,0 +1,176 @@
+include <flat_pack.scad>;
+
+/*
+ * A finger-jointed lasercut box maker OpenSCAD library.
+ *
+ * Inspired by http://boxdesigner.connectionlab.org/
+ */
+module box_2d(box_inner, thickness, tabs, margin=2) {
+  layout_2d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
+    side_a(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+  }
+}
+
+// An assembled version of the box for previewing.
+module box_3d(box_inner, thickness, tabs) {
+  layout_3d(box_inner, thickness) {
+    side_a_top(box_inner, thickness, tabs);
+    side_a(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_b(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+    side_c(box_inner, thickness, tabs);
+  }
+}
+
+// lasercut layout
+module layout_2d(inner, thickness, margin=2) {
+  spacingA = inner[0] + thickness * 2 + margin;
```

Comment:
What are spacingA/B? More descriptive variable names might be a good idea.

Reply:
Good call. Also, spacingB is not used, so I'll remove it.